simpleui-server/src/SuiZMQ.ts

```diff
@@ -0,0 +1,150 @@
+            this.connect(this.port);
+
+            this.socket.on('message', (msg) => {
+                const raw_zmq_data = msg.toString();
+                const zmq_data = SuiData.addXmlStatus(raw_zmq_data);
+                Logger.log(LogLevel.DEBUG, `Recieved ZMQ message at port ${this.port}: ${zmq_data.substring(0, 105)}`);
+                // get response + request from queue
+                let [res, req] = this.http_queue.dequeue();
+                // send response
+                SuiData.sendResponse(req, res, zmq_data);
+            });
+
+            this.socket.on('error', (zmqErr) => {
+                const zmq_data = ServerUtil.getServerError('ZMQ_ERROR', '{{ERROR}}', zmqErr);
+                Logger.log(LogLevel.ERROR, `ZMQ socket at port ${this.port} got error: ${zmqErr}`);
+                // get response + request from queue
+                let [res, req] = this.http_queue.dequeue();
+                // send response
+                SuiData.sendResponse(req, res, zmq_data);
+            });
+
+        } catch (err) {
+            Logger.log(LogLevel.ERROR, `Could not create ZMQ socket at port ${this.port} got error: ${err}`)
+        }
+    }
+
+    connect(port: number) {
+        try {
+            Logger.log(LogLevel.VERBOSE, `ZMQ connecting to --->  tcp://${this.hostname}:${port}`)
+            this.socket.connect(`tcp://${this.hostname}:${port}`)
+
+        } catch (e) {
+            Logger.log(LogLevel.ERROR, `Could not connect to tcp://${this.hostname}:${port}`);
+        }
+    }
+
+    send(msg: string) {
+        try{
+            this.socket.send(msg);
+        } catch (err) {
+            Logger.log(LogLevel.ERROR, `could not send zmq message:\n${msg}`)
+        }
+    }
+
+    close() {
+        if (this.socket.closed === false) {
+            this.socket.close();
+        }
+    }
+
+    set_timeout(ms: number) {
+        try {
+            this.socket.connect_timeout = ms;
+        } catch (err) {
+            Logger.log(LogLevel.ERROR, `COULD NOT SET ZMQ TIMEOUT ${ms}`);
+        }
+    }
+}
+
+
+/**
+ * This class handles the creation of the zmq map 
+ * and the destruction of the sockets
+ */
+export class zmq_wrapper {
+    socket_map: Map<number, ZMQ_Socker_Wrapper>;
+
+    constructor(port_list: Array<number>) {
+        this.socket_map = new Map();
+
+        for (let port_indx = 0; port_indx < port_list.length; port_indx++) {
+            const port = port_list[port_indx];
+            const zmq_wrapper_instance = new ZMQ_Socker_Wrapper('svcmachineapps', port)
+            this.socket_map.set(port, zmq_wrapper_instance);
+        }
+
+        
+        process.on('SIGINT', () => {
+            Logger.log(LogLevel.INFO, 'ZMQ got SIGINT')
+            this.socket_map.forEach((zmq_wrapper_instance, port) => {
+                Logger.log(LogLevel.INFO, `closing zmq port: ${port}`);
+                zmq_wrapper_instance.close();
+            });
+        });
```

Comment:
the callbacks for each of these appear to be identical, you should pull the anonymous arrow function out into a class method called handleApplicationExit() or whatever and just bind that function to those signals.

Reply:
I ended up making the `handleApplicationExit()` method